stko/molecular/topology_extractor/topology_extractor.py

```diff
@@ -0,0 +1,73 @@
+"""
+Topology Extractor
+==================
+
+Class for defining a topology from a molecule and disconnections.
+
+"""
+
+import logging
+
+from ..networkx import Network
+
+logger = logging.getLogger(__name__)
+
+
+class TopologyExtractor:
+    """
+    Extractor of topology definitions from a molecule.
+
+    """
+
+    def extract_topology(
+        self,
+        molecule,
+        atom_ids_to_disconnect,
+    ):
+
+        flat_tuple = []
+        for i in atom_ids_to_disconnect:
+            flat_tuple.append(i[0])
+            flat_tuple.append(i[1])
+        flat_tuple = tuple(set(flat_tuple))
+
+        connected_graphs = self.get_connected_graphs(
+            molecule=molecule,
+            atom_ids_to_disconnect=atom_ids_to_disconnect,
+        )
+
+        # Get centroids.
+        self._centroids = {}
+        self._connectivities = {}
+        for i, cg in enumerate(connected_graphs):
```

Comment:
theres lots of redundant calls to `list(cg)` here. I think you can just use `cg`.

Reply:
Correct, and done.